### PR TITLE
[BSE-4341] Benchmark against Modin/Ray

### PR DESCRIPTION
**Pull request by scott-routledge2**
## Changes included in this PR

Copies bodo implementation, removes jit and adds `import modin.pandas as pd`, also contains config file for launching ray cluster.

## Testing strategy

Running locally on small data to verify output matches, running full set on cluster to measure performance.

Reading Time:  165.34 s
Compute Time: > 3000s

## Checklist
- [ ] Pipelines passed before requesting review. To run CI you must include `[run CI]` in your commit message.
- [x] I am familiar with the [Contributing Guide](https://github.com/bodo-ai/Bodo/blob/main/CONTRIBUTING.md) 
- [x] I have installed + ran pre-commit hooks.

### CODE REVIEW

File: benchmarks/modin-ray/modin-cluster.yaml [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+                      VolumeSize: 500
+                      VolumeType: gp3
+            # Additional options in the boto docs.
+    ray.worker.default:
+        # The minimum number of worker nodes of this type to launch.
+        # This number should be >= 0.
+        min_workers: 3
+        # The maximum number of worker nodes of this type to launch.
+        # This takes precedence over min_workers.
+        max_workers: 3
```
Comment (ehsantn) [inline benchmarks/modin-ray/modin-cluster.yaml:94]: I assume this means 4 total workers including the head node? The head node is utilized for computation I assume?

File: benchmarks/modin-ray/nyc_taxi_preciptation.py [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+    end = time.time()
+    print("Monthly Taxi Travel Times Computation Time: ", end - start)
+
+    start = time.time()
+    monthly_trips_weather.to_parquet("modin_results.pq")
+    return monthly_trips_weather
+
+
+if __name__ == "__main__":
+    result = run_modin()
```
Comment (ehsantn) [inline benchmarks/modin-ray/nyc_taxi_preciptation.py:108]: Looks like `-d small` and `-d large` arguments need added.

**Review by ehsantn — APPROVED** [review]
Thanks @scott-routledge2 . See comments.

File: benchmarks/modin-ray/modin-cluster.yaml (hunk 1 above)
Comment (scott-routledge2) [reply benchmarks/modin-ray/modin-cluster.yaml:94]: Yes

File: benchmarks/modin-ray/nyc_taxi_preciptation.py (hunk 2 above)
Comment (scott-routledge2) [reply benchmarks/modin-ray/nyc_taxi_preciptation.py:115]: Was in the process of removing since we were going to run each system multiple times anyways

**Pull request merged**